backend/src/index.ts

```diff
@@ -288,6 +355,33 @@ app
     }
   });
 
+// Profile data route
+app.post('/api/saveProfileData', async (req, res) => {
+  try {
+    const profile = new ProfileModel(req.body);
+    await profile.save();
+    res.status(201).send('Profile data saved successfully!');
+  } catch (error) {
+    console.error(error);
```

Comment:
Minor critique. This prevents ESLint from emitting warnings relating to this `console` statement
```suggestion
    // eslint-disable-next-line no-console
    console.error(error);
```